.github/workflows/build_test_deploy.yml

```
@@ -1,13 +1,242 @@
+          POSTGRES_PASSWORD: secret_password
+
+      - name: Build webknossos-datastore (sbt)
+        run: sbt -no-colors -DfailOnWarning "project webknossosDatastore" copyMessages compile stage
+
+      - name: Build webknossos-tracingstore (sbt)
+        run: sbt -no-colors -DfailOnWarning "project webknossosTracingstore" copyMessages compile stage
+
+      - name: Lint backend code and check formatting
+        run: .github/not-on-master.sh sbt ";scapegoat; scalafmtCheck; util/scalafmtCheck; webknossosTracingstore/scalafmtCheck; webknossosDatastore/scalafmtCheck"
+
+      - name: Run backend tests
+        run: .github/not-on-master.sh sbt -v "testOnly backend.*"
+
+      - name: Checksum App Dirs
+        run: find app webknossos-datastore/app webknossos-tracingstore/app -type f -exec md5sum {} \; | sort -k 2 | md5sum > app_checksum.txt
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build webknossos docker image
+        run: |
+          docker pull scalableminds/webknossos:$NORMALIZED_BRANCH || true
+          DEV_CACHE=$NORMALIZED_BRANCH docker compose build --pull webknossos
+        env:
+          POSTGRES_URL: jdbc:postgresql://localhost:5432/webknossos
+          POSTGRES_USER: webknossos_user
+          POSTGRES_PASSWORD: secret_password
+
+      - name: Build webknossos-datastore docker image
+        run: docker compose build --pull webknossos-datastore
+
+      - name: Build webknossos-tracingstore docker image
+        run: docker compose build --pull webknossos-tracingstore
+
+        # Enable tmate debugging of manually-triggered workflows if the input option was provided
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
+        if: ${{ github.event_name == 'workflow_dispatch' && inputs.debug_enabled }}
+
+      - name: Start webknossos, datastore, and tracingstore
+        run: |
+          mkdir -p fossildb/data && sudo chmod -R "0777" fossildb/data
+          mkdir -p fossildb/backup && sudo chmod -R "0777" fossildb/backup
+          docker compose up -d webknossos
+          docker compose up -d webknossos-datastore
+          docker compose up -d webknossos-tracingstore
+
+      - name: Run webknossos smoke test
+        uses: ./.github/actions/url-health-check
+        with:
+          url: http://localhost:9000/api/health
+
+      - name: Run webknossos-datastore smoke test
+        uses: ./.github/actions/url-health-check
+        with:
+          url: http://localhost:9090/data/health
+
+      - name: Run webknossos-tracingstore smoke test
+        uses: ./.github/actions/url-health-check
+        with:
+          url: http://localhost:9050/tracings/health
+
+      - name: Run end-to-end tests
+        uses: nick-invision/retry@v2
+        with:
+          timeout_minutes: 10
+          max_attempts: 3
+          command: .github/not-on-master.sh yarn run test-e2e
+        env:
+          POSTGRES_URL: jdbc:postgresql://postgres/webknossos
```

```suggestion
          POSTGRES_URL: jdbc:postgresql://localhost/webknossos
```

Use `localhost` instead of `postgres` seems to work in previous stages